src/pykeen/models/meta/filtered.py

```diff
@@ -0,0 +1,217 @@
+"""Filtered models."""
+
+from typing import Collection, List, Optional, Union
+
+import numpy
+import scipy.sparse
+import torch
+from class_resolver import HintOrType
+from docdata import parse_docdata
+
+from pykeen.evaluation.evaluator import prepare_filter_triples
+
+from ..base import Model
+from ..baseline.utils import get_csr_matrix
+from ...triples.triples_factory import CoreTriplesFactory
+from ...typing import InductiveMode, MappedTriples
+
+__all__ = [
+    "CooccurrenceFilteredModel",
+]
+
+
+@parse_docdata
+class CooccurrenceFilteredModel(Model):
+    """A model which filters predictions by co-occurence.
+
+    ---
+    citation:
+        author: Berrendorf
+        year: 2022
+        link: https://github.com/pykeen/pykeen/pull/943
+        github: pykeen/pykeen
+    """
+
+    head_per_relation: scipy.sparse.csr_matrix
+    tail_per_relation: scipy.sparse.csr_matrix
+    relation_per_head: scipy.sparse.csr_matrix
+    relation_per_tail: scipy.sparse.csr_matrix
+
+    TRAINING_FILL_VALUE: float = -1.0e03
+    INFERENCE_FILL_VALUE: float = float("-inf")
+
+    def __init__(
+        self,
+        *,
+        triples_factory: CoreTriplesFactory,
+        additional_triples: Union[None, MappedTriples, List[MappedTriples]] = None,
+        apply_in_training: bool = False,
+        base: HintOrType[Model] = "rotate",
+        **kwargs,
+    ) -> None:
+        """
+        Initialize the model.
+
+        :param triples_factory:
+            the (training) triples factory; used for creating the co-occurrence counts *and* for instantiating the
+            base model.
+        :param additional_filter_triples:
+            additional triples to use for creating the co-occurrence statistics
+        :param in_training:
+            whether to apply the masking also during training
+        :param base:
+            the base model, or a hint thereof.
+        """
+        # avoid cyclic imports
+        from .. import model_resolver
+
+        # create base model
+        base = model_resolver.make(base, triples_factory=triples_factory, pos_kwargs=kwargs)
+
+        super().__init__(
+            triples_factory=triples_factory,
+            loss=base.loss,
+            predict_with_sigmoid=base.predict_with_sigmoid,
+            random_seed=base._random_seed,
+        )
+        # assign *after* nn.Module.__init__
+        self.base = base
+
+        # index
+        h, r, t = (
+            prepare_filter_triples(
+                mapped_triples=triples_factory.mapped_triples, additional_filter_triples=additional_triples or []
```

Comment:
this function accepts none, I think it should be the same as an empty list, so maybe better to remove `or []`

Reply:
This will emit warnings about not using the filtered setting by Bordes 😅

Reply:
e338da524d7662c5cf0e0eaae879a9893f01a947...e71a0ec893ba95cf19ddf8b587c413033f926145